### PR TITLE
Fix some of the nits in #3214

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -1768,9 +1770,9 @@ completes, the client uses the transport parameters established in the
 handshake.  Not all transport parameters are remembered, as some do not apply to
 future connections or they have no effect on use of 0-RTT.
 
-The definition of new transport parameters ({{new-transport-parameters}}) MUST
-specify whether they MUST, MAY, or MUST NOT be stored for 0-RTT. A client need
-not store a transport parameter it cannot process.
+The definition of new transport parameters ({{new-transport-parameters}}) is
+REQUIRED to specify whether they MUST, MAY, or MUST NOT be stored for 0-RTT. A
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:1774]:
```suggestion
The definition of a new transport parameter ({{new-transport-parameters}}) is
REQUIRED to specify whether it MUST, MAY, or MUST NOT be stored for 0-RTT. A
```
I think the singular reads better, especially with the change in keyword.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -1889,7 +1891,7 @@ confirms that the client received the Initial packet from the server.  Once the
 server has successfully processed a Handshake packet from the client, it can
 consider the client address to have been validated.
 
-Additionally, a server MAY consider the client address valididated if the
+Additionally, a server MAY consider the client address validated if the
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:1894]: 😄

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -3855,9 +3857,7 @@ during connection establishment and when migrating to a new path
 On receiving a QUIC packet with an ECT or CE codepoint, an ECN-enabled endpoint
 that can access the ECN codepoints from the enclosing IP packet increases the
 corresponding ECT(0), ECT(1), or CE count, and includes these counts in
-subsequent ACK frames; see {{generating-acks}} and {{frame-ack}}.  Note that
-this requires being able to read the ECN codepoints from the enclosing IP
-packet, which is not possible on all platforms.
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:3860]: I'm not sure about this one.  IIRC, we were pretty deliberate about adding this text to note that compliance with the ECN guidance was conditional on OS support for doing so.

Comment (MikeBishop) [reply draft-ietf-quic-transport.md:3860]: Yes, added in response to #2156 (PR #2201).

File: draft-ietf-quic-transport.md [hunk 4]
```diff
@@ -1076,8 +1076,8 @@ A zero-length connection ID can be used when a connection ID is not needed to
 route to the correct endpoint. However, multiplexing connections on the same
 local IP address and port while using zero-length connection IDs will cause
 failures in the presence of peer connection migration, NAT rebinding, and client
-port reuse; and therefore MUST NOT be done unless an endpoint is certain that
-those protocol features are not in use.
+port reuse. An endpoint MUST NOT use an IP address and port pair for multiple
+connections unless it is certain that those protocol features are not in use.
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md:1080]: This misses the zero-length connection ID qualification, which is the key point.

File: draft-ietf-quic-transport.md [hunk 5]
```diff
@@ -1232,7 +1232,9 @@ cannot be reverted.
 Valid packets sent to clients always include a Destination Connection ID that
 matches a value the client selects.  Clients that choose to receive
 zero-length connection IDs can use the local address and port to identify a
-connection.  Packets that do not match an existing connection are discarded.
+connection.  Packets that do not match an existing connection, based on
+Destination Connection ID or, if zero-length, local IP address and port,
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md:1236]:
```suggestion
Destination Connection ID or, if this value zero-length, local IP address and port,
```

File: draft-ietf-quic-transport.md (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-quic-transport.md:1774]: Also, MUST > REQUIRED.

**Review by martinthomson — CHANGES_REQUESTED** [review]
What Mike said applies also.

File: draft-ietf-quic-transport.md (hunk 5 above)
Comment (MikeBishop) [reply draft-ietf-quic-transport.md:1236]: "this value **is**"

File: draft-ietf-quic-transport.md (hunk 1 above)
Comment (larseggert) [reply draft-ietf-quic-transport.md:1774]: No, MUST == REQUIRED. (I think you are thinking of RECOMMENDED?)

File: draft-ietf-quic-transport.md (hunk 3 above)
Comment (larseggert) [reply draft-ietf-quic-transport.md:3860]: Am fine with leaving this as is. @gorryfair?

File: draft-ietf-quic-transport.md (hunk 4 above)
Comment (larseggert) [reply draft-ietf-quic-transport.md]: Tried to address this.

File: draft-ietf-quic-transport.md (hunk 3 above)
Comment (gorryfair) [reply draft-ietf-quic-transport.md:3860]: Simply ignoring ECN is not an acceptable position.  My comment here was not a request to remove that. It was a request to clarify what happens on those other platforms: "On those platforms, QUIC will not enable use of ECN for this endpoint.”

Comment (larseggert) [reply draft-ietf-quic-transport.md:3860]: But if the platform doesn't allow access to the ECN bits, the QUIC stack will not be able to use ECN *at all*. So why scope this to an endpoint?

Comment (gorryfair) [reply draft-ietf-quic-transport.md:3860]: OK!  SO what I intended to be clear about was if a platform can't receive ECN marks, then use of ECN can't be enabled to this endpoint.

I too would like to leave the note.

Comment (larseggert) [reply draft-ietf-quic-transport.md:3860]: I added a paragraph to the previous section to clarify this.

Comment (gorryfair) [reply draft-ietf-quic-transport.md:3860]: That new text has resolved this NiT.

File: draft-ietf-quic-transport.md [hunk 6]
```diff
@@ -1768,9 +1772,9 @@ completes, the client uses the transport parameters established in the
 handshake.  Not all transport parameters are remembered, as some do not apply to
 future connections or they have no effect on use of 0-RTT.
 
-The definition of new transport parameters ({{new-transport-parameters}}) MUST
-specify whether they MUST, MAY, or MUST NOT be stored for 0-RTT. A client need
-not store a transport parameter it cannot process.
+The definition of a new transport parameter ({{new-transport-parameters}}) is
+REQUIRED to specify whether it MUST, MAY, or MUST NOT be stored for 0-RTT. A
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md:1776]: This is fine, but I tend to prefer:

```suggestion
The definition of a new transport parameter ({{new-transport-parameters}}) MUST
specify whether that storing the transport parameter for 0-RTT is mandatory, optional, or prohibited. A
```

(Whether an extension uses 2119 language or not is not relevant, the level of requirement is.  And as these keywords are not used in this context in the way described in RFC 2119, this is better.)

**Review by martinthomson — APPROVED** [review]
Nits only.  Thanks for working through these.

File: draft-ietf-quic-transport.md [hunk 7]
```diff
@@ -1768,9 +1772,10 @@ completes, the client uses the transport parameters established in the
 handshake.  Not all transport parameters are remembered, as some do not apply to
 future connections or they have no effect on use of 0-RTT.
 
-The definition of new transport parameters ({{new-transport-parameters}}) MUST
-specify whether they MUST, MAY, or MUST NOT be stored for 0-RTT. A client need
-not store a transport parameter it cannot process.
+The definition of a new transport parameter ({{new-transport-parameters}}) MUST
+specify whether that storing the transport parameter for 0-RTT is mandatory,
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:1776]:
```suggestion
specify whether storing the transport parameter for 0-RTT is mandatory,
```

**Comment by larseggert** [comment]
Looks like this is good to go.

File: draft-ietf-quic-transport.md [hunk 8]
```diff
@@ -3842,6 +3847,10 @@ congestion in the network by setting a codepoint in the IP header of a packet
 instead of dropping it.  Endpoints react to congestion by reducing their sending
 rate in response, as described in {{QUIC-RECOVERY}}.
 
+Note that supporting ECN requires being able to set and read the ECN codepoints
+from the IP headers of packets carrying QUIC. On platforms where this is not
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md:3851]:
```suggestion
from the IP headers of datagrams carrying QUIC packets. On platforms where this is not
```

File: draft-ietf-quic-transport.md [hunk 9]
```diff
@@ -3842,6 +3847,10 @@ congestion in the network by setting a codepoint in the IP header of a packet
...
+possible, QUIC cannot support ECN.
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md:3852]:
```suggestion
 this is not possible, QUIC cannot support ECN.
```

**Pull request merged**